### PR TITLE
Simplify the smoketests that are run by openQA/kanku

### PR DESCRIPTION
**Pull request by hennevogel**

### CODE REVIEW

File: dist/t/spec/features/0040_package_spec.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    fill_in 'name', with: 'ctris'
+    fill_in 'title', with: 'ctris'
+    fill_in 'description', with: 'ctris'
+    click_button('Save changes')
+    expect(page).to have_content("Package 'ctris' was created successfully")
+
+    logout
+  end
+
+  it "should be able to upload files" do
```
Comment (ChrisBr) [inline dist/t/spec/features/0040_package_spec.rb]: What about also testing the interconnect? So branching from interconnect? However, then we need to rely that build.o.o is working and that also the package we branch is not broken ...

File: dist/t/spec/features/0010_authentification_spec.rb [hunk 2]
```diff
@@ -1,17 +1,22 @@
 require "spec_helper"
 
-RSpec.describe "Sign Up & Login" do
-  it "should be able to sign up successfully and logout" do
+RSpec.describe "Authentification" do
```
Comment (hennevogel) [inline dist/t/spec/features/0010_authentification_spec.rb]: https://www.youtube.com/watch?v=zZ3fjQa5Hls&feature=youtu.be&t=91

File: dist/t/spec/features/0040_package_spec.rb (hunk 1 above)
Comment (hennevogel) [inline dist/t/spec/features/0040_package_spec.rb]: See "should be able to branch" :-)

File: dist/t/spec/features/0040_package_spec.rb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    login('Admin','opensuse')
+
+    counter = 100
+    while counter > 0 do
+      visit("/package/show/home:Admin/ctris")
+      # wait for the build results ajax call
+      sleep(5)
+      puts "Refreshed build results, #{counter} retries left."
+      succeed_build = page.all('td', class: 'status_succeeded')
+      if succeed_build.length == 1
```
Comment (hennevogel) [inline dist/t/spec/features/0040_package_spec.rb]: counter = 0 does this right?

**Comment by hennevogel** [comment]
[(8) P3: openQA setup for 2.8](https://trello.com/c/EDt3KCPG/297-8-p3-openqa-setup-for-2-8)

File: dist/t/spec/features/0040_package_spec.rb (hunk 3 above)
Comment (ChrisBr) [inline dist/t/spec/features/0040_package_spec.rb]: Yes of course, but

- I think break is more readable / obvious
- The code goes back to head of the loop for an additional check of the conditional, with break it will just 'break' out of the loop

File: dist/t/spec/features/0010_authentification_spec.rb (hunk 2 above)
Comment (ChrisBr) [inline dist/t/spec/features/0010_authentification_spec.rb]: Nope! According to [Wiktionary](https://en.wiktionary.org/wiki/authentification) it is a misspelling of Authentication (from German Authentifikation). See also [stackexchange](http://english.stackexchange.com/questions/7844/is-authentification-a-real-word)

File: dist/t/spec/features/0040_package_spec.rb (hunk 1 above)
Comment (ChrisBr) [inline dist/t/spec/features/0040_package_spec.rb]: Sorry, missed it!

File: dist/t/spec/features/0010_authentification_spec.rb (hunk 2 above)
Comment (eduardoj) [inline dist/t/spec/features/0010_authentification_spec.rb]: Being this file in `features` subdirectory, wouldn't it be better like this?

`RSpec.feature "Authentification", :type => :feature do`

And also in the other files too.

File: dist/t/spec/features/0010_authentification_spec.rb [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
+  it "should be able to sign up" do
```
Comment (eduardoj) [inline dist/t/spec/features/0010_authentification_spec.rb]: Being this file in `features` subdirectory, wouldn't it be better to replace `it` with `scenario`?

And also in the other `it` lines, too.

File: dist/t/spec/features/0030_project_spec.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
+require "spec_helper"
+
+RSpec.describe "Project" do
+  it "should be able to create" do
+    login('Admin','opensuse')
```
Comment (eduardoj) [inline dist/t/spec/features/0030_project_spec.rb]: I would put this line in a `before` block. And also remove `login` sentence in line 15.

File: dist/t/spec/features/0040_package_spec.rb [hunk 6]
```diff
@@ -0,0 +1,83 @@
+require "spec_helper"
+
+RSpec.describe "Package" do
+
+  it "should be able to create new" do
+    login('Admin','opensuse')
```
Comment (eduardoj) [inline dist/t/spec/features/0040_package_spec.rb]: I would put this line in a `before` block. And also all the other `login` lines in this file.

File: dist/t/spec/features/0030_project_spec.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    click_link('Create Home')
+    expect(page).to have_content("Create New Project")
+    click_button('Create Project')
+    expect(page).to have_content("Project 'home:Admin' was created successfully")
+
+    logout
```
Comment (eduardoj) [inline dist/t/spec/features/0030_project_spec.rb]: I would put this line in an `after` block. And also remove `logout` in line 27.

File: dist/t/spec/features/0040_package_spec.rb [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    end
+    expect(current_path).to eq '/project/show/home:Admin'
+    click_link('Create package')
+    fill_in 'name', with: 'ctris'
+    fill_in 'title', with: 'ctris'
+    fill_in 'description', with: 'ctris'
+    click_button('Save changes')
+    expect(page).to have_content("Package 'ctris' was created successfully")
+
+    logout
```
Comment (eduardoj) [inline dist/t/spec/features/0040_package_spec.rb]: I would put this line in an `after` block. And remove all the `logout` lines in the file.

File: dist/t/spec/features/0030_project_spec.rb (hunk 5 above)
Comment (hennevogel) [inline dist/t/spec/features/0030_project_spec.rb]: Why?

File: dist/t/spec/features/0030_project_spec.rb (hunk 7 above)
Comment (hennevogel) [inline dist/t/spec/features/0030_project_spec.rb]: Why?

File: dist/t/spec/features/0040_package_spec.rb (hunk 8 above)
Comment (hennevogel) [inline dist/t/spec/features/0040_package_spec.rb]: Why? :-)

File: dist/t/spec/features/0030_project_spec.rb (hunk 7 above)
Comment (eduardoj) [inline dist/t/spec/features/0030_project_spec.rb]: Because of the "Don't repeat yourself" principle.

File: dist/t/spec/features/0040_package_spec.rb (hunk 8 above)
Comment (eduardoj) [inline dist/t/spec/features/0040_package_spec.rb]: Because of the "Don't repeat yourself" principle.

File: dist/t/spec/features/0030_project_spec.rb (hunk 5 above)
Comment (eduardoj) [inline dist/t/spec/features/0030_project_spec.rb]: Because of the "Don't repeat yourself" principle.

File: dist/t/spec/spec_helper.rb [hunk 9]
```diff
@@ -1,47 +1,50 @@
...
+  line_number = example.metadata[:line_number]
   screenshot_name = "screenshot-#{filename}-#{line_number}.png"
   screenshot_path = File.join(SCREENSHOT_DIR, screenshot_name)
   page.save_screenshot(screenshot_path)
-  puts meta[:full_description] + "\n Screenshot: #{screenshot_path}"
 end
 
-def obs_login (user,password)
-
+def login(user,password)
```
Comment (bgeuken) [inline dist/t/spec/spec_helper.rb]: @hennevogel I like the idea. What do you say?

File: dist/t/spec/support/capybara.rb [hunk 10]
```diff
@@ -1,16 +1,17 @@
...
-Capybara.default_max_wait_time = 6
-
 Capybara.register_driver :poltergeist do |app|
-  Capybara::Poltergeist::Driver.new(app, debug: false, timeout: 8)
+  Capybara::Poltergeist::Driver.new(app, debug: false, timeout: 60)
 end
 
 Capybara.default_driver = :poltergeist
 Capybara.javascript_driver = :poltergeist
+Capybara.save_path = '/tmp/rspec_screens'
```
Comment (bgeuken) [inline dist/t/spec/support/capybara.rb]: If possible we should use the constant from the spec helper here.

File: dist/t/spec/features/0030_project_spec.rb (hunk 7 above)
Comment (bgeuken) [inline dist/t/spec/features/0030_project_spec.rb]: And the login parts could be moved to a before block

File: dist/t/spec/features/0040_package_spec.rb [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+  end
+
+  scenario "should be able to upload files" do
+    login('Admin')
+    within("div#subheader") do
+      click_link('Home Project')
+    end
+    expect(current_path).to eq '/project/show/home:Admin'
+    click_link('ctris')
+    expect(current_path).to eq '/package/show/home:Admin/ctris'
```
Comment (bgeuken) [inline dist/t/spec/features/0040_package_spec.rb]: If this test is really just about uploading we could directly go to that page instead. Or, if we think it's worth to test this, move it into a shared context.

That said, I am ok to do this in a separate PR.

File: dist/t/spec/features/0040_package_spec.rb [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+    expect(page).to have_content('Package was successfully removed.')
+
+    logout
+  end
+
+  scenario "should be able to successfully build" do
+    login('Admin')
+
+    counter = 100
+    while counter > 0 do
```
Comment (bgeuken) [inline dist/t/spec/features/0040_package_spec.rb]: Could also be written as ``` 100.downto(0) do |counter| ``` :-)

File: dist/t/spec/spec_helper.rb [hunk 13]
```diff
@@ -35,7 +35,7 @@ def take_screenshot(example)
   page.save_screenshot(screenshot_path)
 end
 
-def login(user,password)
+def login(user, password='opensuse')
```
Comment (bgeuken) [inline dist/t/spec/spec_helper.rb]: Please rebase with 97658f9401e8b11d1e16b3ba1a51072df052dde6

**Comment by bgeuken** [comment]
LGTM overall

**Comment by hennevogel** [comment]
@ChrisBr @eduardoj @bgeuken I've tried to incorporate your feedback and hit some more things while doing that (see the [webui] commits). Please review again, sorry.

**Comment by bgeuken** [comment]
LGTM 👍

**Comment by bgeuken** [comment]
@hennevogel Feel free to merge once the rubocop complains got fixed;-)
